qiskit_experiments/library/randomized_benchmarking/rb_experiment.py

```diff
@@ -242,4 +260,37 @@ def _metadata(self):
         for run_opt in ["meas_level", "meas_return"]:
             if hasattr(self.run_options, run_opt):
                 metadata[run_opt] = getattr(self.run_options, run_opt)
+
+        # Set constraints to compute basis gates EPGs from an estimated EPC
+        gate_error_ratio = self.experiment_options.gate_error_ratio
+
+        if gate_error_ratio == "default":
+            # Extract basis gate from transpile option, then backend configuration
+            try:
+                basis_gates = self.transpile_options.get(
+                    "basis_gates", self.backend.configuration().basis_gates
+                )
+                # Do gate error ratio look-up
+                gate_error_ratio = {}
+                for basis_gate in basis_gates:
+                    r_epg = lookup_epg_ratio(basis_gate, self.num_qubits)
+                    if r_epg is None:
+                        continue
+                    gate_error_ratio[(tuple(self.physical_qubits), basis_gate)] = r_epg
+            except AttributeError:
+                # When basis gates is not provided, disable EPG computation
+                warnings.warn(
+                    "The basis gates information is not available. Cannot compute EPGs.",
+                    UserWarning,
+                )
+                gate_error_ratio = False
+
+        def _to_tuple(value):
+            # For JSON serialization. The dict key is not string.
+            if isinstance(value, dict):
+                return tuple(value.items())
```

Comment:
Good catch! I didn't check serialized data. You are correct. I simplified data model because we need to add `qubits` since this is obvious. Then key becomes just a gate name which is JSON serializable.